### ISSUE TITLE
Duplicate notebooks does not work

### ISSUE THREAD

**Issue by raproth** [spark-notebook/spark-notebook#846]
I'm using the current Master (build from spark-notebook.io , 0.8.0-SNAPSHOT-8d5304ca501e856fbda07b0a26a0cdc8b67b47a3 ) and having problems saving my notebook while using the default filesystem storage provider (as given in application.conf)

When I try to duplicate a Notebook (in my case `ClusterTest` which is top-level, i.e. under `notebooks/ClusterTest.snb`) does does nothing, I find in the logs:

```
17/05/03 11:49:58 ERROR play: Cannot invoke the action, eventually got an error: java.nio.file.NoSuchFileException: /home/myuser/spark-notebook-0.8.0-SNAPSHOT-scala-2.11.8-spark-2.1.1-hadoop-2.6.0-cdh5.8.4-with-hive/notebooks/../../notebooks/ClusterTest.snb
17/05/03 11:49:58 ERROR application:

! @73pgff1ho - Internal server error, for (POST) [/api/contents/] ->

@73pgff1ho: Execution exception in null:null
        at play.api.Application$class.handleError(Application.scala:296)
        at play.api.DefaultApplication.handleError(Application.scala:402)
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320)
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3$$anonfun$applyOrElse$4.apply(PlayDefaultUpstreamHandler.scala:320)
        at scala.Option.map(Option.scala:146)
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:320)
        at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$3.applyOrElse(PlayDefaultUpstreamHandler.scala:316)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:346)
        at scala.concurrent.Future$$anonfun$recoverWith$1.apply(Future.scala:345)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:32)
        at play.api.libs.iteratee.Execution$trampoline$.execute(Execution.scala:46)
        at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:40)
        at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:248)
        at scala.concurrent.Promise$class.complete(Promise.scala:55)
        at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:153)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:23)
        at akka.dispatch.TaskInvocation.run(AbstractDispatcher.scala:40)
        at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:397)
        at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: java.nio.file.NoSuchFileException:  /home/myuser/spark-notebook-0.8.0-SNAPSHOT-scala-2.11.8-spark-2.1.1-hadoop-2.6.0-cdh5.8.4-with-hive/notebooks/../../notebooks/ClusterTest.snb
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102)
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107)
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214)
        at java.nio.file.Files.newByteChannel(Files.java:361)
        at java.nio.file.Files.newByteChannel(Files.java:407)
        at java.nio.file.Files.readAllBytes(Files.java:3152)
        at notebook.io.FileSystemNotebookProviderConfigurator$FileSystemNotebookProvider$$anonfun$get$1.apply(FileSystemNotebookProviderConfigurator.scala:46)
        at notebook.io.FileSystemNotebookProviderConfigurator$FileSystemNotebookProvider$$anonfun$get$1.apply(FileSystemNotebookProviderConfigurator.scala:46)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
        ... 4 more

```

**Comment by raproth** [comment]
The problem is a `NoSuchFileException`

he tries to read:
`/home/myuser/spark-notebook-0.8.0-SNAPSHOT-scala-2.11.8-spark-2.1.1-hadoop-2.6.0-cdh5.8.4-with-hive/notebooks/../../notebooks/ClusterTest.snb`

but the correct path would be:
`/home/myuser/spark-notebook-0.8.0-SNAPSHOT-scala-2.11.8-spark-2.1.1-hadoop-2.6.0-cdh5.8.4-with-hive/notebooks/../notebooks/ClusterTest.snb`

**Comment by andypetrella** [comment]
mmmmh, that is strange yes. Do you use a different conf file? 
Do you run the spark notebook server from the bin folder? or base folder?

You can always update the path to the notebook dir using the configuration otherwise :-S

**Comment by raproth** [comment]
I just unpacked the zip-file with the build, did not change anything except the port in `conf/application.ini`

I started the notebook using `bin/spark-notebook` but also tried to start it from within `bin`, makes no difference.

I've had no problems with SN version 0.7.0 so far... strange

**Comment by raproth** [comment]
Found a solution, had to change:

``` 
notebooks {
    # Server dir (containing notebook files)
    dir = ./notebooks
    dir = ${?NOTEBOOKS_DIR}
  }
```

This also solves https://github.com/spark-notebook/spark-notebook/issues/847
to 

``` 
notebooks {
    # Server dir (containing notebook files)
    dir = notebooks
    dir = ${?NOTEBOOKS_DIR}
  }
```

**Comment by vidma** [comment]
P.S. I think I used `export NOTEBOOKS_DIR=/absolute/path/to-notebooks-dir` when testing.
will check.

**Comment by vidma** [comment]
@raproth, thanks for reporting this. it should be fixed in #849

**Comment by raproth** [comment]
@vidma thanks for fixing!
